Tribler/Test/Core/Modules/MetadataStore/test_channel_metadata.py

```diff
@@ -285,6 +266,82 @@ def test_commit_channel_torrent(self):
         self.assertTrue(channel.commit_channel_torrent())
         self.assertFalse(channel.commit_channel_torrent())
 
+    @db_session
+    def test_recursive_commit_channel_torrent(self):
+        status_types = [NEW, UPDATED, TODELETE, COMMITTED]
+
+        def all_status_combinations():
+            result = []
+            for card in range(0, len(status_types) + 1):
+                result.extend(list(combinations(status_types, card)))
+            return result
+
+        def gen_coll(parent, collection_status, contents_statuses, recurse=False):
+            chan = self.mds.CollectionNode(
+                title=parent.title + '->child_new_nonempty', origin_id=parent.id_, status=collection_status
+            )
+            for s in contents_statuses:
+                self.mds.TorrentMetadata(infohash=random_infohash(), origin_id=chan.id_, status=s)
+                if recurse:
+                    for status in status_types:
+                        gen_coll(chan, status, [NEW])
+            return chan
+
+        def gen_chan(recurse=False, status=NEW):
```

Comment:
Can you just name this `generate_channel`?

Reply:
Ok